packages/cloud_firestore/lib/src/firestore.dart

```diff
@@ -52,4 +60,70 @@ class Firestore {
     assert(path != null);
     return new DocumentReference._(this, path.split('/'));
   }
+
+  /// Runs a set of atomic database operations.
+  Future<Map<String, dynamic>> runTransaction(TransactionHandler transactionHandler,
+      {Duration timeout: const Duration(seconds: 5)}) async {
+    assert(timeout.inMilliseconds > 0, 'Transaction timeout must e more than 0 milliseconds');
+    final int transactionId = _transactionHandlerId++;
+    _transactionHandlers[transactionId] = transactionHandler;
+    // Execute operations on the native side.
+    final Map<String, dynamic> result = await channel.invokeMethod('Firestore#runTransaction',
+        <String, dynamic>{'transactionId': transactionId, 'transactionTimeout': timeout.inMilliseconds});
+    if (result != null) {
+      return result;
+    } else {
+      return <String, dynamic>{};
+    }
+  }
+}
+
+typedef Future<Map<String, dynamic>> TransactionHandler(Transaction tx);
+
+class Transaction {
+  MethodChannel _channel;
+  int _transactionId;
+
+  Transaction(this._transactionId) {
+    _channel = Firestore.channel;
+  }
+
+  Future<DocumentSnapshot> get(DocumentReference documentReference) async {
+    return _channel.invokeMethod('Transaction#get', <String, dynamic>{
+      'transactionId': _transactionId,
+      'path': documentReference.path
+    }).then((dynamic result) {
```

Comment:
I think it would be more readable with `await`

Reply:
Done!